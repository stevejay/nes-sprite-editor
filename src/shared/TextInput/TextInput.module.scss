@import "styles/_base.scss";

$button-v-padding: 10;

.input {
  -webkit-appearance: none;
  -moz-appearance: none;
  width: 100%;
  height: rem($button-v-padding + $button-v-padding + 16 + 2);
  padding: 0 rem(12);
  font-family: inherit;
  font-size: rem(16);
  vertical-align: middle;
  border: 1px solid $md-grey-500;
  border-radius: rem(3);
  outline: 0;
  background-color: $md-white;
  line-height: 1.5rem;
}

.input:disabled {
  background-color: $md-grey-200;
  border-color: $md-grey-400;
  color: $md-grey-500;
  cursor: not-allowed;
}

.input:focus {
  @include focus-ring;
}
