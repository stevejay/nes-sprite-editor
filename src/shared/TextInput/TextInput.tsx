import React from "react";
import styles from "./TextInput.module.scss";

export type Props = {
  type?: "text";
  value: string;
  disabled?: boolean;
  id?: string;
  name?: string;
  autosOff?: boolean;
  onChange: (value: string) => void;
};

const TextInput = ({
  type = "text",
  value,
  disabled = false,
  id,
  name,
  autosOff,
  onChange
}: Props) => (
  <input
    id={id}
    name={name}
    className={styles.input}
    type={type}
    value={value}
    disabled={disabled}
    onChange={event => onChange(event.target.value)}
    autoCapitalize={autosOff ? "off" : undefined}
    autoComplete={autosOff ? "off" : undefined}
    autoCorrect={autosOff ? "off" : undefined}
    spellCheck={autosOff ? false : undefined}
  />
);

export default TextInput;
