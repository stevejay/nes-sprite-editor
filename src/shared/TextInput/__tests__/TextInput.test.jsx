import React from "react";
import { noop } from "lodash";
import { render, cleanup } from "react-testing-library";
import TextInput from "../TextInput";

afterEach(cleanup);

test("displays correctly", async () => {
  const { container } = render(
    <TextInput
      value="the value"
      id="the-id"
      name="the-name"
      autosOff
      onChange={noop}
    />
  );
  expect(container).toMatchSnapshot();
});
