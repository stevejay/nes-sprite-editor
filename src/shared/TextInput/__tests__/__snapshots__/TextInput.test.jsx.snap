// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correctly 1`] = `
<div>
  <input
    autocapitalize="off"
    autocomplete="off"
    autocorrect="off"
    class="input"
    id="the-id"
    name="the-name"
    spellcheck="false"
    type="text"
    value="the value"
  />
</div>
`;
